### DISCUSSION TITLE
Enhanced RTMP

### DISCUSSION THREAD

**Discussion by nums** [OvenMediaLabs/OvenMediaEngine#1167] [Ideas]
A good news to take in consideration ?

https://github.com/obsproject/obs-studio/pull/8522

**Comment by alexthedamager** [comment]
This is very interesting!

**Comment by naanlizard** [comment]
Very interested in trying this, I assume it's not supported by ome and op though?

**Comment by getroot** [comment]
Thanks for sharing good information.

I'm also very interested in this. I think I can start working on officially supporting AV1 and HEVC in OME soon. This will be a task to support the new codecs in all modules such as Input (Provider)/Encoding/Output (Publisher). This is a very large task and Enhanced RTMP support will obviously be a subtask.

And I'm more interested in AV1. Apple started supporting AV1 a while ago and it is expected that Safari and mobile will also support it. However, I suspect that HEVC will not be able to be used in WebRTC in the future, so depending on the global spread of AV1, HEVC support may become a very low priority. Of course, if Chrome starts supporting HEVC in WebRTC, the priorities may change.

**Comment by basisbit** [comment]
I think the major use case where support for Enhanced RTMP would be needed is the RTMP Push provider because then OME can be used for common 4K video streaming cases where you cant just stream a 30Mb/s video over the locations internet upload.

Hardware accelerated AV1 decoding support is already somewhat wide spread, in devices of the past 4 years and it is mostly patent / royalty fee free which means it will be usable for client side too. (HEVC/h265 is mostly dead on client side for live streaming because of patent and royalty fees hell)

Hardware accelerated AV1 encoding support just started to be common for any new graphics card generation around middle of last year. The new Intel & Nvidia and AMD GPUs all support AV1 encoding, but almost everything before that and all low-energy chipsets don't support AV1 encoding yet.

Thus, for live streaming 4K video, the neartime way to go seems to be: use H265/HEVC over SRT to OvenMediaEngine, from there you send a copy of the media transport stream over Enhanced RTMP to YouTube, and you transcode the stream on the OME server to AVC/x264 to then deliver it to WebRTC/LLHLS/HLS viewers (this needs Intel or Nvidia GPU hardware acceleration for decoding the h265 stream. The single-threaded decoding which OME does can not handle anything more than HD without GPU hardware acceleration)
